mne/utils/mixin.py

```diff
@@ -501,8 +575,107 @@ def decimate(self, decim, offset=0, verbose=None):
         else:
             self._decim_slice = decim_slice
         self._set_times(self._raw_times[self._decim_slice])
+        self._update_first_last()
+        return self
+
+    # Overridden method signature does not match call...
+    def time_as_index(self, times, use_rounding=False):  # lgtm
+        """Convert time to indices.
+
+        Parameters
+        ----------
+        times : list-like | float | int
+            List of numbers or a number representing points in time.
+        use_rounding : bool
+            If True, use rounding (instead of truncation) when converting
+            times to indices. This can help avoid non-unique indices.
+
+        Returns
+        -------
+        index : ndarray
+            Indices corresponding to the times supplied.
+        """
+        from ..source_estimate import _BaseSourceEstimate
+        if isinstance(self, _BaseSourceEstimate):
+            sfreq = 1. / self.tstep
+        else:
+            sfreq = self.info['sfreq']
+        index = (np.atleast_1d(times) - self.times[0]) * sfreq
+        if use_rounding:
+            index = np.round(index)
+        return index.astype(int)
+
+    def _handle_tmin_tmax(self, tmin, tmax):
+        """Convert seconds to index into data.
+
+        Parameters
+        ----------
+        tmin : int | float | None
+            Start time of data to get in seconds.
+        tmax : int | float | None
+            End time of data to get in seconds.
```

Comment:
Should this support an `include_tmax` argument similar to `crop`? 
Related to #10372

Reply:
Perhaps eventually yes... I'll just add a note to #10372 that we should tackle it at the same time as that ideally